pkg/controllermanager/controller/project/project_stale_control.go

```diff
@@ -172,64 +149,73 @@ func (r *projectStaleReconciler) reconcile(ctx context.Context, project *gardenc
 	return r.gardenClient.Delete(ctx, project)
 }
 
-func (r *projectStaleReconciler) projectInUseDueToShoots(namespace string) (bool, error) {
-	shootList, err := r.shootLister.Shoots(namespace).List(labels.Everything())
-	return len(shootList) > 0, err
+func (r *projectStaleReconciler) projectInUseDueToShoots(ctx context.Context, namespace string) (bool, error) {
+	shootList := &gardencorev1beta1.ShootList{}
```

Comment:
Please use metadata-only lists with `client.Limit(1)` in the funcs that only check for existence of any respective object.
Otherwise we will stress client and server quite heavily with large list calls.

Reply:
Yes, makes sense, please check it out: ac864f8